sdk/dotnet/AdmissionRegistration/V1/MutatingWebhookConfiguration.cs

```diff
@@ -0,0 +1,94 @@
+    /// <summary>
+    /// MutatingWebhookConfiguration describes the configuration of and admission webhook that
+    /// accept or reject and may change the object.
+    /// </summary>
+    public partial class MutatingWebhookConfiguration : Pulumi.CustomResource {
+        /// <summary>
+        /// APIVersion defines the versioned schema of this representation of an object. Servers
+        /// should convert recognized schemas to the latest internal value, and may reject
+        /// unrecognized values. More info:
+        /// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+        /// </summary>
+        [Output("apiVersion")]
+        public Output<string> ApiVersion { get; private set; } = null!;
+
+        /// <summary>
+        /// Kind is a string value representing the REST resource this object represents. Servers
+        /// may infer this from the endpoint the client submits requests to. Cannot be updated. In
+        /// CamelCase. More info:
+        /// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+        /// </summary>
+        [Output("kind")]
+        public Output<string> Kind { get; private set; } = null!;
+
+        /// <summary>
+        /// Standard object metadata; More info:
+        /// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata.
+        /// </summary>
+        [Output("metadata")]
+        public Output<Types.Outputs.Meta.V1.ObjectMeta> Metadata { get; private set; } = null!;
+
+        /// <summary>
+        /// Webhooks is a list of webhooks and the affected resources and operations.
+        /// </summary>
+        [Output("webhooks")]
+        public Output<Types.Outputs.AdmissionRegistration.V1.MutatingWebhook[]> Webhooks { get; private set; } = null!;
+
+
+        /// <summary>
+        /// Create a MutatingWebhookConfiguration resource with the given unique name, arguments, and options.
+        /// </summary>
+        ///
+        /// <param name="name">The unique name of the resource</param>
+        /// <param name="args">The arguments used to populate this resource's properties</param>
+        /// <param name="options">A bag of options that control this resource's behavior</param>
+        public MutatingWebhookConfiguration(string name, Types.Inputs.AdmissionRegistration.V1.MutatingWebhookConfiguration? args = null, CustomResourceOptions? options = null)
+            : base("kubernetes:admissionregistration.k8s.io/v1:MutatingWebhookConfiguration", name, SetAPIKindAndVersion(args), MakeResourceOptions(options, ""))
+        {
+        }
+
+        private static ResourceArgs? SetAPIKindAndVersion(Types.Inputs.AdmissionRegistration.V1.MutatingWebhookConfiguration? args) {
+            if (args != null) {
+                args.ApiVersion = "admissionregistration.k8s.io/v1";
+                args.Kind = "MutatingWebhookConfiguration";
+            }
```

Comment:
i don't mind 'heavyweght' extra work we do that the user won't see.  it's not like they'd ever be aware of this or feel any perf hit here (compared to the IO and costs of the actual K work).  So i'm ok with just generating internally and doing the copies.

Reply:
I actually was slightly wrong about this.  In TypeScript, we do expose these properties on the input interfaces, but we then also always replace whatever values are set with fixed values.  So it's mostly the same as what we currently do in C#.  It's probably slightly "wrong" in both, but I think I'll leave as-is for now.  

The one thing that's slightly different is that TS copies the fields into a new object instead of modifying the args object passed in by the user.  I'm okay doing that for now here - though it's something we could potentially improve on in the future without a breaking change here.